pkg/slo-controller/noderesource/plugins/util/util.go

```diff
@@ -181,11 +187,12 @@ func GetPodMetricUsage(info *slov1alpha1.PodMetricInfo) corev1.ResourceList {
 	return GetResourceListForCPUAndMemory(info.PodUsage.ResourceList)
 }
 
-func GetHostAppHPUsed(resourceMetrics *framework.ResourceMetrics) corev1.ResourceList {
+func GetHostAppHPUsed(resourceMetrics *framework.ResourceMetrics, pluginName string) corev1.ResourceList {
 	hostAppHPUsed := util.NewZeroResourceList()
 	for _, hostAppMetric := range resourceMetrics.NodeMetric.Status.HostApplicationMetric {
-		if hostAppMetric.Priority == extension.PriorityBatch || hostAppMetric.Priority == extension.PriorityFree {
-			// only consider higher priority usage for batch allocatable
+		if (pluginName == midPluginName && hostAppMetric.Priority == extension.PriorityMid) ||
```

Comment:
it is not recommended to hard-coding the plugin name in the util func

Reply:
we can simply compare if the priority class is higher

Reply:
fixed.